### PR TITLE
chore(sto-bro): add filter locations e2e test

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Add an e2e test for filtering locations

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/e2e/features/ui/components/storage/storage-browser/filter-locations.feature [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    When I type my "email" with status "CONFIRMED"
+    Then I type my password
+    Then I click the "Sign in" button
+    Then I see the first button containing "private"
+    When I see input with placeholder "Filter folders and files" and type "pu"
+    Then I click the "Submit" button
+    Then I see the first button containing "public"
+    Then I do not see the button containing "private"
+    When I click the button containing "Clear search"
+    Then I see the first button containing "private"
```
Comment (jordanvn) [inline packages/e2e/features/ui/components/storage/storage-browser/filter-locations.feature:17]: Needs EOF newline?

File: packages/e2e/cypress/integration/common/shared.ts [hunk 2]
```diff
@@ -288,6 +288,12 @@ Then('I see the button containing {string}', (name: string) => {
   }).should('exist');
 });
 
+Then('I do not see the button containing {string}', (name: string) => {
+  cy.findByRole('button', {
+    name: new RegExp(`${escapeRegExp(name)}`, 'i'),
+  }).should('not.exist');
+});
+
```
Comment (jordanvn) [inline packages/e2e/cypress/integration/common/shared.ts:296]: It looks like this functionality already exists as ["I do not see the {string} button"](https://github.com/aws-amplify/amplify-ui/blob/feat-storage-browser/main/packages/e2e/cypress/integration/common/shared.ts#L305)

File: packages/e2e/features/ui/components/storage/storage-browser/filter-locations.feature (hunk 1 above)
Comment (calebpollman) [inline packages/e2e/features/ui/components/storage/storage-browser/filter-locations.feature:17]:
```suggestion
 Then I see the first button containing "private"

```

**Review by jordanvn — APPROVED** [review]
LGTM, assuming we'll maintain the `public` and `private` directories for testing. Surprised these steps didn't already exist, but confirmed that I couldn't find equivalent functionality

**Review by HuiSF — APPROVED** [review]

**Pull request merged**